src/test/kotlin/com/terraformation/backend/accelerator/db/CohortStoreTest.kt

```diff
@@ -234,6 +269,83 @@ class CohortStoreTest : DatabaseTest(), RunsAsUser {
           CohortPhaseUpdatedEvent(cohortId, CohortPhase.Phase1FeasibilityStudy))
     }
 
+    @Test
+    fun `upserts cohort modules`() {
+      every { user.canUpdateCohort(any()) } returns true
+
+      val cohortId = insertCohort()
+      val module1 = insertModule()
+      val module2 = insertModule()
+
+      insertCohortModule(
+          cohortId,
+          module1,
+          startDate = LocalDate.of(2024, 1, 1),
+          endDate = LocalDate.of(2024, 1, 3))
+
+      assertEquals(
+          listOf(
+              CohortModulesRow(
+                  cohortId, module1, LocalDate.of(2024, 1, 1), LocalDate.of(2024, 1, 3))),
+          cohortModulesDao.findAll(),
+          "Cohort modules before update.")
+
+      store.update(cohortId) {
+        it.copy(
+            modules =
+                listOf(
+                    CohortModuleModel(
+                        cohortId,
+                        module1,
+                        LocalDate.of(2024, 2, 1),
+                        LocalDate.of(2024, 2, 3),
+                    ),
```

Comment:
Maybe insert this one beforehand too, so the test covers the three cases of delete, keep, and add rather than just delete and add.